### DISCUSSION TITLE
Change Colom DataType

### DISCUSSION THREAD

**Discussion by jjcomg6** [JoshClose/CsvHelper#1870] [Q&A]
How To Change Column Data Type From Number To Text Before Export CSV File?

**Comment by jjcomg6** [comment]
My Problem Is With This Values For Example 0921456 In Export Changed To Number 921456 !

**Reply by vernetk** [reply]
Hi,

Sorry but if your datatype is an integer in your datatable column then csvhelper has nothing to do with losing the "0" character in front of it.
It is a standard rule, the 0 in front of a numeric are removed.

Can you suggest an example to illustrate your words?



`public static void Main(string[] args)
        {
            var dt = new DataTable();
            dt.Columns.Add("Id", typeof(int));
            dt.Columns.Add("Name", typeof(string));
            dt.Columns.Add("Number", typeof(int));

            var row = dt.NewRow();
            row["Id"] = 1;
            row["Name"] = "0921456";
            row["Number"] = 0921456;
            dt.Rows.Add(row);

            row = dt.NewRow();
            row["Id"] = 2;
            row["Name"] = "10921456";
            row["Number"] = 10921456;
            dt.Rows.Add(row);

            string result;
            using (var writer = new StringWriter())
            using (var csv = new CsvWriter(writer, CultureInfo.InvariantCulture))
            {
                foreach (DataColumn dc in dt.Columns)
                {
                    csv.WriteField(dc.ColumnName);
                }
                csv.NextRecord();

                foreach (DataRow dr in dt.Rows)
                {
                    foreach (DataColumn dc in dt.Columns)
                    {
                        csv.WriteField(dr[dc]);
                    }
                    csv.NextRecord();
                }

                result = writer.ToString();
            }
            Console.WriteLine($"Result: {result}");
            Console.ReadLine();
        }`

![image](https://user-images.githubusercontent.com/5453308/135819464-df21b932-c427-46f7-bd8a-3b199dbc9305.png)
